src/gulpfile.js

```diff
@@ -186,47 +186,74 @@ gulp.task('chat-styles', ['chat-styles-handlebars', 'chat-styles-js'], function(
 
 // Compile and concatenate handlebars files for all chat styles.
 gulp.task('chat-styles-handlebars', function() {
-    // TODO: Process all available styles, not only the default one.
-    var stylePath = config.chatStylesPath + '/default';
-
-    return gulp.src(stylePath + '/templates_src/client_side/**/*.handlebars')
-        .pipe(handlebars({
-            // Use specific version of Handlebars.js
-            handlebars: handlebarsEngine
-        }))
-        .pipe(wrapHandlebarsTemplate())
-        .pipe(concat('templates.js'))
-        .pipe(uglify({preserveComments: 'some'}))
-        .pipe(header(config.compiledTemplatesHeader))
-        .pipe(gulp.dest(stylePath + '/templates_compiled/client_side'));
+    var promises = [];
+    fs.readdir(config.chatStylesPath, function(err, list){
+        if(err) return done(err);
+        list.filter(function(path){
+            return fs.lstatSync(config.chatStylesPath + "/" + path).isDirectory();
+        }).map(function(dir){
+            promises.push(new Promise(function(resolve, reject){
```

Comment:
`Array.prototype.push` is not needed. Just return a promise from (that's `Array.prototype.map` is designed for).